src/PandoraMonitoring.cc

```diff
@@ -1207,6 +1326,182 @@ void PandoraMonitoring::InitializeEve(Char_t transparency)
+ 
+    pTEveWindowSlot = TEveWindow::CreateWindowInTab(m_pEveManager->GetBrowser()->GetTabRight());
+    pTEveWindowPack = pTEveWindowSlot->MakePack();
+    pTEveWindowPack->SetElementName("V View");
+    pTEveWindowPack->SetHorizontal();
+    pTEveWindowPack->SetShowTitleBar(kFALSE);
+    pTEveWindowPack->NewSlot()->MakeCurrent();
+    m_p2DVView = m_pEveManager->SpawnNewViewer("2D V View", "");
+    this->AddScenes(m_p2DVView, m_p2DVEventScene, m_p2DVGeometryScene, TGLViewer::kCameraOrthoXOZ, 1);
+
+    // Combined 2d Views
+    pTEveWindowSlot = TEveWindow::CreateWindowInTab(m_pEveManager->GetBrowser()->GetTabRight());
+    pTEveWindowPack = pTEveWindowSlot->MakePack();
+    pTEveWindowPack->SetElementName("2D Views");
+    pTEveWindowPack->SetHorizontal();
+    pTEveWindowPack->SetShowTitleBar(kFALSE);
+    pTEveWindowPack->NewSlot()->MakeCurrent();
+
+    m_p2DWView = m_pEveManager->SpawnNewViewer("2D W View", "");
+    this->AddScenes(m_p2DWView, m_p2DWEventScene, m_p2DWGeometryScene, TGLViewer::kCameraOrthoXOZ, 1);
+
+    pTEveWindowPack->SetShowTitleBar(kFALSE);
+    pTEveWindowPack->NewSlot()->MakeCurrent();
+    m_p2DUView = m_pEveManager->SpawnNewViewer("2D U View", "");
+    this->AddScenes(m_p2DUView, m_p2DUEventScene, m_p2DUGeometryScene, TGLViewer::kCameraOrthoXOZ, 1);
+
+    pTEveWindowPack->SetShowTitleBar(kFALSE);
+    pTEveWindowPack->NewSlot()->MakeCurrent();
+    m_p2DVView = m_pEveManager->SpawnNewViewer("2D V View", "");
+    this->AddScenes(m_p2DVView, m_p2DVEventScene, m_p2DVGeometryScene, TGLViewer::kCameraOrthoXOZ, 1);
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+void PandoraMonitoring::AddScenes(TEveViewer *pTEveViewer, TEveScene *pTEveEventScene, TEveScene *pTEveGeometryScene, TGLViewer::ECameraType camera, int axisType)
+{
+    pTEveViewer->AddScene(pTEveGeometryScene);
+    pTEveViewer->AddScene(pTEveEventScene);
+    pTEveViewer->GetGLViewer()->SetCurrentCamera(camera);
+    pTEveViewer->GetGLViewer()->ColorSet().Background().SetColor(kWhite);
+
+    int currentAxisType(0);
+    bool axisDepthTest(false);
+    bool referenceOn(false);
+    double referencePosition[3] = {0.0, 0.0, 0.0};
+    pTEveViewer->GetGLViewer()->GetGuideState(currentAxisType, axisDepthTest, referenceOn, referencePosition);
+    pTEveViewer->GetGLViewer()->SetGuideState(axisType, axisDepthTest, referenceOn, referencePosition);
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+void PandoraMonitoring::InitializeLArTPCs(TGeoVolume *pMainDetectorVolume, TGeoMedium *pLArTPCMedium, Char_t transparency)
+{
+    const LArTPCMap &larTPCMap(m_pPandora->GetGeometry()->GetLArTPCMap());
+
+    int color(4);
+    for (auto &iter : larTPCMap)
+    {
+        const std::string name(iter.first);
```

Comment:
Probably want a reference to the string in the map, rather than taking a copy.

Reply:
Good point.  Will make this change.